mypy/test/data/check-classes.test

```diff
@@ -1418,3 +1418,215 @@ C(0)
 C(arg=0)
 [out]
 
+
+-- Type[C]
+-- -------
+
+
+[case testTypeUsingTypeCBasic]
+from typing import Type
+class User: pass
+class ProUser(User): pass
+def new_user(user_class: Type[User]) -> User:
+    return user_class()
```

Comment:
Test calling with invalid arguments to `__init__`. Also test a class with non-default `__init__` and call constructor with both valid and invalid arguments.